src/GitVersionCore/OutputVariables/VariableProvider.cs

```diff
@@ -100,5 +94,24 @@ static void PromoteNumberOfCommitsToTagNumber(SemanticVersion semanticVersion)
             semanticVersion.BuildMetaData.CommitsSinceVersionSource = semanticVersion.BuildMetaData.CommitsSinceTag ?? 0;
             semanticVersion.BuildMetaData.CommitsSinceTag = null;
         }
+
+        static void CheckAndFormatString<T>(string formatString, T source,  string defaultValue, string formatVarName, out string formattedString)
```

Comment:
Neat, I like this. But is there a reason why the `formattedString` is an `out` parameter instead of a return value?

Reply:
No preferences. Matter of taste

Reply:
Addressed in the latest commit